homeassistant/util/secrets.py

```diff
@@ -0,0 +1,132 @@
+
+
+def load_secrets_yaml(config_path, filename='secrets.yaml'):
+    """Try to load secrets.yaml."""
+    from homeassistant.config import load_yaml_config_file
+    from homeassistant.exceptions import HomeAssistantError
+    global SECRET_DICT
+    try:
+        SECRET_DICT = load_yaml_config_file(
+            os.path.join(os.path.dirname(config_path), filename))
+        _LOGGER.info(filename + ' loaded')
+        if 'debug' in SECRET_DICT and SECRET_DICT['debug']:
+            _LOGGER.setLevel(logging.DEBUG)
+    except FileNotFoundError:
+        SECRET_DICT = None
+    except HomeAssistantError:
+        _LOGGER.error('Could not load %s', filename)
+        SECRET_DICT = None
+
+
+def check_secrets_on_start(hass):
+    """Check secrets on homeassistant startup."""
+    def hass_started(event):
+        """Callback when hass started."""
+        check_secrets(True)
+    from homeassistant.const import EVENT_HOMEASSISTANT_START
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, hass_started)
+
+
+def check_secrets(log_warning=False):
+    """Check for missing or any unused secrets."""
+    missing = [sname for sname, ssrc in HISTORY.items()
+               if ssrc == SSource.notfound]
+    unused = []
+    if SECRET_DICT:
+        unused = [sname for sname in SECRET_DICT.keys()
+                  if sname not in HISTORY and sname != 'debug']
+    if log_warning:
+        if len(unused) > 0:
+            _LOGGER.warning('Unused secrets [secrets.yaml]: ' +
+                            ', '.join(unused))
+        if len(missing) > 0:
+            _LOGGER.warning('Missing secrets: ' + ', '.join(missing))
+    return {'missing': missing, 'unused': unused}
+
+
+def _cmd_line():
+    """Set / Get secrets through cmdline."""
+    if not keyring:
+        _LOGGER.error("keyring needs to be installed (pip install keyring)")
+        return
+    import argparse
+    parser = argparse.ArgumentParser(
+        description="Manage secrets in the keyring.",
+        epilog="Alternative use: keyring [set|get|del] homeassistant name")
+    parser.add_argument('action', help="The action [get|set|del]")
+    parser.add_argument('name', help="The name of the secret")
+    arguments = parser.parse_args()
+    if arguments.action == 'set':
+        from getpass import getpass
+        secret = getpass("Secret for '{}': ".format(arguments.name))
+        if isinstance(secret, str) and len(secret) > 0:
+            keyring.set_password(_SECRET_NAMESPACE, arguments.name, secret)
+        else:
+            print("Not set")
+    elif arguments.action == 'del':
+        try:
+            keyring.delete_password(_SECRET_NAMESPACE, arguments.name)
+        except keyring.errors.PasswordDeleteError:
+            print("Secret not found.")
+    elif arguments.action == 'get':
+        print(keyring.get_password(_SECRET_NAMESPACE, arguments.name))
+    else:
+        print('Invalid action: {}'.format(arguments.action))
+
+if __name__ == "__main__":
+    _cmd_line()
```

Comment:
I don't want to sprinkle command line helpers throughout the app. This should move to `__main__.py`. That file is getting very full so we could consider making a new subfolder with command line commands